dj/sql/parsing/ast.py

```diff
@@ -0,0 +1,491 @@
+
+    def alias_or_name(self) -> str:
+        """
+        get the name or alias of the node
+        """
+        if len(self.parents) == 1:
+            parent = tuple(self.parents)[0]
+            if isinstance(parent, Alias):
+                return parent.name
+        return self.name
+
+
+class Operation(Expression):
+    """a type to overarch types that operate on other expressions"""
+
+
+class UnaryOpKind(Enum):
+    """the accepted unary operations"""
+
+    Plus = "+"  # pylint: disable=C0103
+    Minus = "-"  # pylint: disable=C0103
+    Not = "NOT"  # pylint: disable=C0103
+
+
+@dataclass(eq=False)
+class UnaryOp(Operation):
+    """an operation that operates on a single expression"""
+
+    op: UnaryOpKind  # pylint: disable=C0103
+    expr: Expression
+
+    def __hash__(self) -> int:
+        return hash((UnaryOp, self.op))
+
+
+class BinaryOpKind(Enum):
+    """the accepted binary operations"""
+
+    And = "AND"  # pylint: disable=C0103
+    Or = "OR"  # pylint: disable=C0103
+    Eq = "="  # pylint: disable=C0103
+    NotEq = "<>"  # pylint: disable=C0103
+    Gt = ">"  # pylint: disable=C0103
+    Lt = "<"  # pylint: disable=C0103
+    GtEq = ">="  # pylint: disable=C0103
+    LtEq = "<="  # pylint: disable=C0103
+    BitwiseOr = "|"  # pylint: disable=C0103
+    BitwiseAnd = "&"  # pylint: disable=C0103
+    BitwiseXor = "^"  # pylint: disable=C0103
+    Multiply = "*"  # pylint: disable=C0103
+    Divide = "/"  # pylint: disable=C0103
+    Plus = "+"  # pylint: disable=C0103
+    Minus = "-"  # pylint: disable=C0103
+    Modulo = "%"  # pylint: disable=C0103
+
+
+@dataclass(eq=False)
+class BinaryOp(Operation):
+    """represents an operation that operates on two expressions"""
+
+    left: Expression
+    op: BinaryOpKind  # pylint: disable=C0103
+    right: Expression
+
+    def __hash__(self) -> int:
+        return hash((BinaryOp, self.op))
+
+
+@dataclass(eq=False)
+class Between(Operation):
+    """
+    a between statement
+    """
+
+    expr: Expression
+    low: Expression
+    high: Expression
+
+    def __hash__(self) -> int:
+        return hash((Between, self.low, self.high))
+
+
+@dataclass(eq=False)
+class Case(Expression):
+    """a case statement of branches"""
+
+    conditions: List[Expression]
+    else_result: Optional[Expression]
+    operand: Optional[Expression]
+    results: List[Expression]
+
+    def __hash__(self) -> int:
+        return id(self)
+
+
+@dataclass(eq=False)
+class Function(Named, Operation):
+    """represents a function used in a statement"""
+
+    args: List[Expression]
+
+    def __hash__(self) -> int:
+        return hash(Function)
+
+
+@dataclass(eq=False)  # type: ignore
+class Value(Expression):
+    """base class for all values number, string, boolean"""
+
+    value: Union[str, bool, float, int]
+
+
+@dataclass(eq=False)
+class Number(Value):  # Number
```

Comment:
super nit: The comment doesn't seem needed here or for the next couple of classes.

Reply:
how can you be sure the class represents a Number without that comment?